inox2d/src/params.rs

```diff
@@ -191,8 +193,9 @@ impl Puppet {
 
 	pub fn set_param(&mut self, param_name: &str, val: Vec2) {
 		let param = self
-			.parameters
+			.param_names
 			.get(param_name)
+			.map(|pn| self.parameters.get(pn).expect("param index mismatch {pn}"))
 			.unwrap_or_else(|| panic!("No parameter named: {}", param_name));
```

Comment:
Ehh yeah I wasn't paying attention to this so it might already be a Result in the other branch ? I'll check and copy
The inner error is a panic error though, the index having a reference to a non-existent value